tests/misc_env/install_prereq.py

```diff
@@ -373,29 +373,19 @@ def registry_login(ceph, distro_ver):
         ceph.exec_command(cmd="sudo systemctl restart docker", long_running=True)
 
     config = get_cephci_config()
-    registries = [
-        {
-            "registry": "registry.redhat.io",
-            "user": config["cdn_credentials"]["username"],
-            "passwd": config["cdn_credentials"]["password"],
+    registries = {
+        "registry.redhat.io": {
+            "username": config["cdn_credentials"]["username"],
+            "password": config["cdn_credentials"]["password"],
         }
-    ]
-
-    if (
-        config.get("registry_credentials")
-        and config["registry_credentials"]["registry"] != "registry.redhat.io"
-    ):
-        registries.append(
-            {
-                "registry": config["registry_credentials"]["registry"],
-                "user": config["registry_credentials"]["username"],
-                "passwd": config["registry_credentials"]["password"],
-            }
-        )
+    }
     auths = {}
-    for r in registries:
-        b64_auth = base64.b64encode(f"{r['user']}:{r['passwd']}".encode("ascii"))
-        auths[r["registry"]] = {"auth": b64_auth.decode("utf-8")}
+    if config.get("registry_credentials"):
+        registries.update(config.get("registry_credentials"))
+
+    for k, v in registries.items():
+        b64_auth = base64.b64encode(f"{v['username']}:{v['password']}".encode("ascii"))
+        auths[k] = {"auth": b64_auth.decode("utf-8")}
```

Comment:
If someone hasn't updated their .cephci.yaml based on new format, this part would fail right ?
Can we have some check around the new format and throw user understandable error to user if .cephci.yaml is not updated ?

Reply:
Added exception block for error and provided the details Vasi